### DISCUSSION TITLE
How to load a plugin in github

### DISCUSSION THREAD

**Discussion by ittaysw** [zdharma-continuum/zinit#586] [Q&A] [answered]
This might be trivial for an experienced zinit/omz user, but I'm not sure what is the best way to use https://github.com/zpm-zsh/autoenv via zinit. Tried to use `zinit snippet https://github.com/zpm-zsh/autoenv/blob/master/autoenv.plugin.zsh` and if fails since zinit didn't download the `functions` folder (I think). Should I clone the entire repo? Where (as if it is an omz plugin)? Will it be updated?

**Comment by ittaysw** [comment]
I figured I could do the following, is there a way not to do it with 'svn' (and then 'trunk')? 

```
zinit ice svn pick"autoenv.plugin.zsh"  
zinit snippet https://github.com/zpm-zsh/autoenv/trunk
```

**Reply by pschmitt** [reply, marked as answer]
```zsh
zinit load zpm-zsh/colors
zinit load zpm-zsh/autoenv
```

**Reply by ittayd** [reply]
This works. Thanks!. Can you explain a bit more about the anatomy? Does it just assume the argument is a path on github?

**Reply by pschmitt** [reply]
yes, the second arg is a GitHub repo

For more info I'd suggesting getting through the README :)

**Reply by ittayd** [reply]
I just did. It may be obvious to you, but load is defined to receive plg-spec, but I couldn't find what plg-spec is (or plugin-spec or 'plugin spec'). Also, not clear what is plugin "loading" vs "sourcing" (which is what snippet does). Also the OMZ prefixes and others are kind of confusing. Can I do `zinit load oh-my-zsh/oh-my-zsh/plugins/git` instead of `zinit snippet OMZP::git`? Why can't I do `zinit load OMZP::git` ? 

The framework is amazing in its capabilities and it is evident it has a lot of features to to a lot of effort being put into it.
